src/test/unit/app/controller/courts/PostcodeController.ts

```diff
@@ -0,0 +1,622 @@
+import {mockRequest} from '../../../utils/mockRequest';
+import {mockResponse} from '../../../utils/mockResponse';
+import {CSRF} from '../../../../../main/modules/csrf';
+import {PostcodesController} from '../../../../../main/app/controller/courts/PostcodesController';
+import {AreaOfLaw} from '../../../../../main/types/AreaOfLaw';
+import {CourtType} from '../../../../../main/types/CourtType';
+import {familyAreaOfLaw} from '../../../../../main/enums/familyAreaOfLaw';
+
+describe('PostcodeController', () => {
+
+  let mockApi: {
+    getPostcodes: () => Promise<string[]>,
+    addPostcodes: () => Promise<string[]>,
+    getCourts: () => Promise<object[]>,
+    deletePostcodes: () => Promise<object>,
+    movePostcodes: () => Promise<object[]>,
+    getCourtAreasOfLaw: () => Promise<AreaOfLaw[]>,
+    getCourtCourtTypes: () => Promise<CourtType[]>};
+
+  const testSlug = 'plymouth-combined-court';
+  const getPostcodeData = ['PL1', 'PL2', 'PL3', 'PL11 1YY', 'PL1 1', 'PL 1'];
+  const getPostcodeInput = 'PL1,PL2,PL3,PL11 1YY,PL1 1,PL 1';
+  const getDeletedPostcodes = ['PL1', 'PL2', 'PL3'];
+  const getMovedPostcodes = ['PL11 1YY', 'PL1 1'];
+  const newPostcodes = 'PL4,PL5,PL6';
+  const getPostcodes: () => string[] = () => getPostcodeData;
+  const apiCourtTypesInput: CourtType[] = [
+    { id: 11420, name: 'Crown Court', code: 446 },
+    { id: 11419, name: 'County Court', code: 296 },
+    { id: 11417, name: 'Family Court', code: null }
+  ];
+  const apiCourtTypesInputInvalid: CourtType[] = [
+    { id: 11417, name: 'Family Court', code: null }
+  ];
+  const apiAreasOfLawInput: AreaOfLaw[] = [
+    { id: 1, name:familyAreaOfLaw.bankruptcy},
+    { id: 2, name:familyAreaOfLaw.moneyClaims},
+    { id: 3, name:familyAreaOfLaw.housing}
+  ];
+  const courtTypesBodyInput = 'Crown_Court,County_Court,Family_Court';
+  const areasOfLawBodyInput = familyAreaOfLaw.bankruptcy + ',' + familyAreaOfLaw.moneyClaims + ',' + familyAreaOfLaw.housing;
+  const courtTypesMethodOutput = ['Crown_Court', 'County_Court', 'Family_Court'];
+  const courtTypesMethodOutputInvalid = ['Family_Court'];
+  const areasOfLawMethodOutput = [familyAreaOfLaw.bankruptcy,
+    familyAreaOfLaw.moneyClaims.replace(' ', '_'), familyAreaOfLaw.housing];
+
+  const controller = new PostcodesController();
+
+  beforeEach(() => {
+    mockApi = {
+      getPostcodes: async (): Promise<string[]> => getPostcodes(),
+      addPostcodes: async (): Promise<string[]> => newPostcodes.split(','),
+      getCourts: async (): Promise<object[]> => [],
```

Comment:
Should we add some courts here? We'd need to update some of the test expectations.

Reply:
I think the assertion would simply be that what the api returns, is what we get back in the expected section, so it doesn't matter too much. Happy to add in a few courts though if you think that'd be better! Let me know :D